components/supervisor/pkg/terminal/terminal_test.go

```diff
@@ -49,17 +47,16 @@ func TestTerminals(t *testing.T) {
 				t.Fatal("no terminal")
 			}
 			stdoutOutput := bytes.NewBuffer(nil)
-			go io.Copy(stdoutOutput, terminal.Stdout.Listen())
+			done := make(chan struct{})
+			go func() {
+				io.Copy(stdoutOutput, terminal.Stdout.Listen())
+				done <- struct{}{}
+			}()
 
-			for _, stdin := range test.Stdin {
-				terminal.PTY.Write([]byte(stdin + "\r\n"))
-				time.Sleep(test.Delay)
-			}
+			command := strings.Join(test.Stdin, " && ") + " && exit\r\n"
+			terminal.PTY.Write([]byte(command))
 
-			err = terminalService.Mux.Close(resp.Alias)
-			if err != nil {
-				t.Fatal(err)
-			}
+			<-done
```

Comment:
I tried to use `Process.Wait` but there were a race sometime, just waiting for end of file seems to be more reliable